src/v4/lib/HybridOrderLib.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {InputToken, OutputToken} from "../../base/ReactorStructs.sol";
+import {OrderInfo} from "../base/ReactorStructs.sol";
+import {OrderInfoLib} from "./OrderInfoLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {PriceCurveLib, PriceCurveElement} from "lib/tribunal/src/lib/PriceCurveLib.sol";
+import {FixedPointMathLib} from "solady/utils/FixedPointMathLib.sol";
+
+/// @notice Cosigner data for hybrid auction orders
+struct HybridCosignerData {
+    uint256 auctionTargetBlock;
+    uint256[] supplementalPriceCurve;
+}
+
+/// @notice Input tokens for hybrid auction
+/// @dev if exact-in, scale down from maxAmount
+/// @dev if exact-out, input amount is fixed at maxAmount
+struct HybridInput {
+    ERC20 token;
+    uint256 maxAmount;
+}
+
+/// @notice Output tokens for hybrid auction
+/// @dev if exact-in, scale up from minAmount
+/// @dev if exact-out, output amount is fixed at minAmount
+struct HybridOutput {
+    address token;
+    uint256 minAmount;
+    address recipient;
+}
+
+/// @notice Hybrid auction order combining Dutch decay and priority gas auctions
+struct HybridOrder {
+    OrderInfo info;
+    address cosigner;
+    HybridInput input;
+    HybridOutput[] outputs;
+    uint256 auctionStartBlock;
```

Comment:
Nit: I'm curious why you use `auctionStartBlock` for the order but `auctionTargetBlock` for the cosigner override. Perhaps we can have the cosigner's be `auctionStartBlockOverride` for consistency?

Reply:
following the same lingo we use in priority orders